specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/scheduledActions.json

```diff
@@ -0,0 +1,1001 @@
+            "description": "May be used to filter scheduled actions by properties/viewId. Supported operator is 'eq'.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ScheduledActionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.CostManagement/scheduledActions": {
+      "get": {
+        "tags": [
+          "ScheduledActions"
+        ],
+        "operationId": "ScheduledActions_ListByScope",
+        "description": "List all shared scheduled actions within the given scope.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/rest/api/cost-management/"
+        },
+        "x-ms-examples": {
+          "ScheduledActionsListByScope": {
+            "$ref": "./examples/scheduledActions-list-shared.json"
+          },
+          "ScheduledActionsListByScopeFilterByViewId": {
+            "$ref": "./examples/scheduledActions-listWithFilter-shared.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/scopeParameter"
+          },
+          {
+            "$ref": "common-types.json#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter scheduled actions by properties/viewId. Supported operator is 'eq'.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ScheduledActionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.CostManagement/scheduledActions/{name}": {
+      "put": {
+        "tags": [
+          "ScheduledActions"
+        ],
+        "operationId": "ScheduledActions_CreateOrUpdate",
+        "description": "Create or update a private scheduled action.",
+        "x-ms-examples": {
+          "CreateOrUpdatePrivateScheduledAction": {
+            "$ref": "./examples/scheduledAction-createOrUpdate-private.json"
+          }
```

Comment:
Why isn't this at a resource group scope? You're getting RPC violation linter errors because its at the tenant level.

Reply:
The API is supported at both tenant scope and Subscription/ResourceGroup/BillingAccount etc. scopes

The API at tenant scope is used for creating/managing "private" scheduled actions that nobody other than the creator/owner has access to. Scheduled actions created at ARM RBAC scopes and Billing scopes are accessible to others that have access to the corresponding scope.

In the swagger, we represent ARM scopes by {scope} in the path for ex: /{scope}/providers/Microsoft.CostManagement/scheduledActions

We do this in other versions and API as well, not sure why the linter is flagging this API.